Sources/GPX/OAGPXDocument.mm

```diff
@@ -808,6 +811,25 @@ - (OAGPXTrackAnalysis*) getAnalysis:(long)fileTimestamp
     return g;
 }
 
+- (OAGPXTrackAnalysis*) getCachedAnalysis:(long)fileTimestamp
```

Comment:
The `OAGPXDocument` is immutable. It is does not make sense to implement `getCachedAnalysis` here.
It should be done at `OAGPXMutableDocument` as overridden `getAnalysis` method. Also it should be similar to java code, ie via `modifiedTime` field.
The `modifiedTime` should be set at any method which modifies document like `addWpt`, etc.
At `OAGPXMutableDocument.h` `modifiedTime` should be readonly.